core/embed/rust/src/ui/model_mercury/component/vertical_menu.rs

```diff
@@ -0,0 +1,154 @@
+        // TODO: this is just POC
+        let mut buttons_vec = VerticalMenuButtons::new();
+        let [opt1, opt2, opt3] = options;
+        let [icon1, icon2, icon3] = icons;
+
+        buttons_vec.push(
+            Button::with_icon_and_text(IconText::new(opt1, icon1))
+                .styled(theme::button_vertical_menu()),
+        );
+        buttons_vec.push(
+            Button::with_icon_and_text(IconText::new(opt2, icon2))
+                .styled(theme::button_vertical_menu()),
+        );
+
+        buttons_vec.push(
+            Button::with_icon_and_text(IconText::new(opt3, icon3))
+                .styled(theme::button_vertical_menu_orange()),
+        );
+        Self::new(buttons_vec)
+    }
+}
+
+impl<T> Component for VerticalMenu<T>
+where
+    T: AsRef<str>,
+{
+    type Msg = VerticalMenuChoiceMsg;
+
+    fn place(&mut self, bounds: Rect) -> Rect {
+        // VerticalMenu is supposed to be used in Frame, the remaining space is just
+        // enought to fit 3 buttons separated by thin bars
+        let height_bounds_expected = 3 * MENU_BUTTON_HEIGHT + 2 * MENU_SEP_HEIGHT;
+        assert!(bounds.height() == height_bounds_expected);
+
+        self.area = bounds;
+        let mut remaining = bounds;
+        for i in 0..N_ITEMS {
+            let (area_button, new_remaining) = remaining.split_top(MENU_BUTTON_HEIGHT);
+            self.buttons[i].place(area_button);
+            remaining = new_remaining;
+            if i < N_SEPS {
+                let (area_sep, new_remaining) = remaining.split_top(MENU_SEP_HEIGHT);
+                self.areas_sep.push(area_sep);
+                remaining = new_remaining;
+            }
+        }
+
+        self.area
+    }
+
+    fn event(&mut self, ctx: &mut EventCtx, event: Event) -> Option<Self::Msg> {
+        for (i, button) in self.buttons.iter_mut().enumerate() {
+            if let Some(ButtonMsg::Clicked) = button.event(ctx, event) {
+                return Some(VerticalMenuChoiceMsg::Selected(i));
+            }
+        }
+        None
+    }
+
+    fn paint(&mut self) {
+        // TODO remove when ui-t3t1 done
+    }
+
+    fn render<'s>(&'s self, target: &mut impl Renderer<'s>) {
+        // render buttons separated by thin bars
+        for button in self.buttons.iter() {
+            button.render(target);
+        }
+        for area in self.areas_sep.iter() {
+            Bar::new(*area)
+                .with_thickness(MENU_SEP_HEIGHT)
+                .with_fg(theme::GREY_EXTRA_DARK)
+                .render(target);
+        }
+    }
+
+    #[cfg(feature = "ui_bounds")]
+    fn bounds(&self, sink: &mut dyn FnMut(Rect)) {
+        sink(self.area);
+    }
+}
+
+#[cfg(feature = "ui_debug")]
+impl<T> crate::trace::Trace for VerticalMenu<T>
+where
+    T: AsRef<str>,
+{
+    fn trace(&self, t: &mut dyn crate::trace::Tracer) {
+        // TODO: implement
+        t.component("VerticalMenu");
+        // t.child("inner", &self.inner);
```

Comment:
Could be something like
```suggestion
        t.in_list("buttons", |button_list| {
            for &button in self.buttons {
                button_list.child(button)
            }
        });
```

Reply:
https://github.com/trezor/trezor-firmware/pull/3599/commits/5664655e7aaf7b92db44c1e60bac06e9d3b56df0